### PR TITLE
Initial Golang verified catalog repo setup

### PR DESCRIPTION
**Pull request by QuanZhang-William**
This commit adds the required files to meet [tektoncd project requirements]( https://github.com/tektoncd/community/blob/main/process.md#project-requirements):

- CONTRIBUTING.md
-  DEVELOPMENT.md
- OWNERS
- code-of-conduct.md
- README.md

The contents will be filled in later PRs

The OWNERS file content is migrated from: https://github.com/tektoncd/catalog/blob/main/OWNERS

### CODE REVIEW

**Comment by QuanZhang-William** [comment]
Part of https://github.com/tektoncd-catalog/golang/issues/3

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1 @@
+TODO: add contributing documentation
```
Comment (vdemeester) [inline CONTRIBUTING.md]: We should adapt the one from `tektoncd/catalog` for this repo I think.

**Review by vdemeester — APPROVED** [review]

**Comment by dibyom** [comment]
Do we want to use a .github repository for the templates? So that newly created projects can just have it automatically? Similar to https://github.com/tektoncd/.github/tree/main/.github ?

File: CONTRIBUTING.md (hunk 1 above)
Comment (QuanZhang-William) [reply CONTRIBUTING.md]: Yeah, I will use it as a template. But there are some changes we need to make; for example, using git based versioning and the new automated testing setup.

Will add the details in later PRs if you don't mind 😁

**Comment by QuanZhang-William** [comment]
> Do we want to use a .github repository for the templates? So that newly created projects can just have it automatically? Similar to https://github.com/tektoncd/.github/tree/main/.github ?

Interesting, I didn't know there is org level .github folder! Will update the PR to use the template.

**Comment by QuanZhang-William** [comment]
> Do we want to use a .github repository for the templates? So that newly created projects can just have it automatically? Similar to https://github.com/tektoncd/.github/tree/main/.github ?

@dibyom @jerop @vdemeester @vinamra28 . I have created the [.github](https://github.com/tektoncd-catalog/.github) and cleaned up the content in this PR. Can you please take a look?

**Review by vinamra28 — APPROVED** [review]

**Pull request merged**